front/components/vaults/VaultSideBarMenu.tsx

```diff
@@ -431,33 +434,87 @@ const VaultDataSourceViewItem = ({
   item,
   owner,
   vault,
+  node,
 }: {
   item: DataSourceViewType;
   owner: LightWorkspaceType;
   vault: VaultType;
+  node?: DataSourceViewContentNode;
 }): ReactElement => {
   const router = useRouter();
+  const [isExpanded, setIsExpanded] = useState(false);
 
-  const LogoComponent = getConnectorProviderLogoWithFallback(
-    item.dataSource.connectorProvider,
-    FolderIcon
-  );
-  const dataSourceViewPath = `/w/${owner.sId}/vaults/${vault.sId}/categories/${item.category}/data_source_views/${item.sId}`;
+  const { isNodesLoading, nodes } = useDataSourceViewContentNodes({
+    dataSourceView: item,
+    owner,
+    parentId: node ? node?.internalId : undefined,
+    viewType: "documents",
+  });
+
+  // Load the currently selected node from router.query.parentId
+  const {
+    nodes: [selected],
+  } = useDataSourceViewContentNodes({
+    dataSourceView: router.query.parentId ? item : undefined,
```

Comment:
Does this work? Shouldn't this be in its own `useEffect()` to be updated when the router changes?

Reply:
Yes, it works - not sure to understand why/where we would need an effect. This is recalculated on every render. The useDataSourceViewContentNodes is actually already an effect which will update the state when the query is done.